api/api_controller.go

```diff
@@ -902,11 +902,12 @@ func (c *Controller) ObjectsStatObjectHandler() objects.StatObjectHandler {
 
 		// serialize entry
 		obj := &models.ObjectStats{
-			Checksum:  entry.Checksum,
-			Mtime:     entry.CreationDate.Unix(),
-			Path:      params.Path,
-			PathType:  models.ObjectStatsPathTypeObject,
-			SizeBytes: entry.Size,
+			Checksum:        entry.Checksum,
+			Mtime:           entry.CreationDate.Unix(),
+			Path:            params.Path,
+			PhysicalAddress: entry.PhysicalAddress,
```

Comment:
Our use of the `PhysicalAddress` internally is kind of weird: it might be a fully qualified URI (i.e. `s3://foo/bar/baz`) or it might be a non-qualified string (i.e. `bar/baz`). The later is what happens when you write an object through the API or S3 Gateway. The former is what happens when you do `lakefs import`.

The idea is that if it isn't a URI, we append it to the repository's storage namespace, otherwise we pass it to the block adapter as is. If we don't want API clients to have to reinvent this logic, we can simply always return the full URI. This means wrapping every returned physical address with `block.ResolveNamespace`. I suggest we do that.

Reply:
Done.

Reply:
And added an issue to achieve the same with lakectl.